### ISSUE TITLE
torch.nn.functional.log_softmax on Arm CPU gives partly NaN results

### ISSUE THREAD

**Issue by ulucsahin** [pytorch/pytorch#55663]
## 🐛 Bug
torch.nn.functional.log_softmax gives partly nan results on Raspberry Pi 4 with 64bit Arm Cortex-A72 processor. Executing exactly same code on x86 processors (tested both on Ubuntu and Windows on multiple machines) gives correct results.

Following line causes the problem on Arm cpu:
`input_ = torch.nn.functional.log_softmax(input_, dim=-1)`

Using an alternate softmax implementation solves the issue:
```
from scipy.special import log_softmax
input_= input_.cpu().detach().numpy()
input_= log_softmax(input_, axis=-1)
input_= torch.from_numpy(input_)
```

## To Reproduce

1. Install PyTorch on Arm based device such as Raspberry Pi
2. Execute torch.nn.functional.log_softmax on CPU

## Expected behavior
An output tensor without NaN values should be obtained.

## Environment

Collecting environment information...
PyTorch version: 1.7.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.2 LTS (aarch64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: version 3.16.3

Python version: 3.8 (64-bit runtime)
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.20.1
[pip3] pytorch-lightning==1.0.8
[pip3] torch==1.7.1
[pip3] torch-stft==0.1.4
[pip3] torchtext==0.6.0
[pip3] torchvision==0.2.2.post3
[conda] Could not collect

## Additional context

None

**Comment by malfet** [comment]
@ulucsahin did you build torch yourself or download the binary from PYPA?
And you please share a full minimal reproducer, as I was not able to repro using following script on arm64:
```
import torch

print(torch.__version__)
for i in range(1000):
  x = torch.rand(3, 64, 74)
  y = torch.nn.functional.log_softmax(x, dim=-1)
  if torch.isnan(y.sum()):
      print("Ieee!")
print("That's all")
```

**Comment by ulucsahin** [comment]
In that case, maybe this is only happening in certain range of values. I was using NVidia's NeMo framework and I encountered the issue when trying to to Speech Recognition (ASR). I will try to post more details on monday (I don't have access to the computer right now).

I downloaded binaries for 1.7.1 and also tried compiling myself for 1.9 version.

**Comment by lakshmi-speak** [comment]
@malfet Any word on this? Seeing this to be an issue on PyTorch Mobile Android 1.12.1 and 1.13.1 on ARM processors.
